client/src/App.tsx

```diff
@@ -13,21 +13,44 @@ import {
 } from '@src/pages';
 import Animator from '@components/Animator';
 import Verification from '@components/Verification';
+import { getUserLocationAPI } from '@apis/user';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import { authAtom } from '@stores/AuthRecoil';
+import { locationAtom } from '@stores/ActionInfoRecoil';
 
 const App: React.FC = () => {
+  const Auth = useRecoilValue(authAtom);
+  const setLocation = useSetRecoilState(locationAtom);
+
+  useEffect(() => {
+    (async () => {
+      if (Auth?.id) {
+        const locationList = await getUserLocationAPI(Auth.id);
+        const { location1Id, location1Name, location2Id, location2Name } =
+          locationList[0];
+        const newLocation = [
+          { id: location1Id, region: location1Name },
+          { id: location2Id, region: location2Name },
+        ];
+
+        setLocation(newLocation.filter(({ id, region }) => id && region));
+      }
+    })();
+  }, [Auth?.id, setLocation]);
+
```

Comment:
```suggestion
  const {
    isLoading,
    isError,
    data: location,
  } = useQuery<TLocation>(
    `detail-${productId}`,
    async () => {
      if (Auth?.id) {
        return await getUserLocationAPI(Auth.id);
      return undefined
     },
    {
      onSuccess(data) {
      if(data)
        setLocation(newLocation.filter(({ id, region }) => id && region));
      },
      refetchOnWindowFocus: false,
      retry: 0,
    },
  );

  useEffect(() => {
    (async () => {
      if (Auth?.id) {
        const locationList = await getUserLocationAPI(Auth.id);
        const { location1Id, location1Name, location2Id, location2Name } =
          locationList[0];
        const newLocation = [
          { id: location1Id, region: location1Name },
          { id: location2Id, region: location2Name },
        ];

        setLocation(newLocation.filter(({ id, region }) => id && region));
      }
    })();
  }, [Auth?.id, setLocation]);

```
위와 같은 방법을 사용하면 `react-query`로도 recoil 상태를 수정할 수 있습니다!
전체적으로 `location`값을 이용하는 만큼 loading 처리는 중요한 것 같아요!

Reply:
알려주신 코멘트 이용해서 react-query를 이용해서 비동기 작업과 리코일 작업을 수행헀습니다.!! 좋은 의견 감사합니다!